osu.Game.Rulesets.Mania/UI/ColumnFlow.cs

```diff
@@ -54,17 +58,28 @@ public ColumnFlow(StageDefinition stageDefinition)
                 columns.Add(new Container<TContent> { RelativeSizeAxes = Axes.Y });
         }
 
-        private ISkinSource currentSkin;
+        private ISkinSource currentSkin = null!;
+
+        private readonly Bindable<ManiaMobilePlayStyle> mobilePlayStyle = new Bindable<ManiaMobilePlayStyle>();
 
         [BackgroundDependencyLoader]
-        private void load(ISkinSource skin)
+        private void load(ISkinSource skin, ManiaRulesetConfigManager? rulesetConfig)
         {
             currentSkin = skin;
 
+            rulesetConfig?.BindWith(ManiaRulesetSetting.MobilePlayStyle, mobilePlayStyle);
+            mobilePlayStyle.BindValueChanged(_ => updateMobileSizing());
+
             skin.SourceChanged += onSkinChanged;
             onSkinChanged();
         }
 
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            updateMobileSizing();
```

Comment:
I'm pretty sure you'll want this to be called on invalidation (drawsize or similar). Mobile devices don't guarantee that the window size will never change, and currently there doesn't seem to be a flow to recalculate things when/if it does.